lib/elixir/lib/module/checker.ex

```diff
@@ -1,36 +1,26 @@
 defmodule Module.Checker do
   def verify(module_map, _binary) do
-    check_definitions(Enum.reverse(module_map.definitions), module_map)
-    collect_warnings()
-  end
+    state = :maps.with([:module, :file, :compile_opts], module_map)
```

Comment:
Not having the `Map` functions will be annoying. Maybe we should add `map.ex` to the list of precompiled modules? Then we can use `Map` instead of `:maps` here, in the ParallelCompiler, and in the Unicode modules.